Pspcl/Pspcl.Services/StockService.cs

```diff
@@ -506,8 +522,83 @@ public void UpdateStockMaterial(List<List<int>> selectedRowsToDelete)
             }
             _dbcontext.SaveChanges();
             
-        }       
+        }
+
+        public string UploadImageToAzure(IFormFile file)
+        {
+            string fileExtension = Path.GetExtension(file.FileName);
+            using MemoryStream fileUploadStream = new MemoryStream();
+            {
+                file.CopyTo(fileUploadStream);
+                fileUploadStream.Position = 0;
+                BlobContainerClient blobContainerClient = new BlobContainerClient(
+                    _azureOptions.ConnectionString,
+                    _azureOptions.Container);
+
+                var uniqueName = Guid.NewGuid().ToString() + fileExtension;
+
+                try
+                {
+                    //azure package is required for below line
+                    BlobClient blobClient = blobContainerClient.GetBlobClient(uniqueName);
+                    blobClient.Upload(fileUploadStream, new BlobUploadOptions()
+                    {
+                        HttpHeaders = new BlobHttpHeaders
+                        {
+                            ContentType = "image/bitmap"
+                        }
+                    }, cancellationToken: default);
+                
+                }
+                catch (RequestFailedException ex)
+                {
+                    // An exception occurred during the upload
+                    Console.WriteLine("Upload failed. Error message: " + ex.Message);
+                }
+
+                return uniqueName;
+            }
+           
+        }
+        public string DownloadFileFromBlob(string fileName)
+        {
+            CloudStorageAccount account = CloudStorageAccount.Parse("DefaultEndpointsProtocol=https;AccountName=pspclstorage;AccountKey=b4bgeAkYvbYW1W50h03HzZ5B4HWS71fmLgdCwMCTif1gfOmJ8no9eewFjL2oTONwV0kz+NkG0owT+AStlsTshQ==;EndpointSuffix=core.windows.net");
+            CloudBlobClient blobClient = account.CreateCloudBlobClient();
+            CloudBlobContainer container = blobClient.GetContainerReference("pspcl-images");
+            CloudBlob blob = container.GetBlobReference(fileName);
+
+            int counter = 0;
+            string fileNameWithoutExtension = Path.GetFileNameWithoutExtension(fileName);
+            string fileExtension = Path.GetExtension(fileName);
+            string localFilePath = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.UserProfile), "Downloads", fileName);
```

Comment:
File/Folder location which is common should be configurable and should come from AppSettings.

Reply:
Here, are you talking about the localFilePath ? if yes, then:
It is being created dynamically, but in appsettings, we can declare the constants only, so I am not getting how it could be done, can you please help me out in this?

Reply:
Environment.SpecialFolder.UserProfile      is specified at runtime, 
"Downloads"  only is static here, so it can be fetched from appsettings file,
fileName     is also dynamic, which we get as a parameter.

Reply:
Yes, I am talking about Downloads folder.